llarp/service/protocol.cpp

```diff
@@ -92,6 +96,32 @@ namespace llarp
       return bencode_end(buf);
     }
 
+    std::optional<std::vector<byte_t>>
+    ProtocolMessage::MaybeEncodeAuthInfo() const
```

Comment:
would throwing be better here? returning `nullopt` on all errors seems like it will make it impossible for the caller to know what went wrong (which might matter in debugging if nothing else).

Reply:
Agree on this - `std::optional` is conceptually not meant for errors, but rather for optional values where empty means you can still go ahead.